### ISSUE TITLE
Fix Request: false positive in AvoidDuplicateAssignmentsInCases

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#168]
Assignment to a boolean should be an exception since it can only contain 2 values.
Example:
```
class Foo {
    private static final Pattern NUM_PAT = Pattern.compile("^[0-9]+$");
    private static final Pattern LET_PAT = Pattern.compile("^[A-z]+$");

    private boolean formatIsValid(String format, String fieldValue) {
        Matcher matcher;
        boolean formatIsValid;
        switch (format) {
            case NUMERIC:
                matcher = numericPattern.matcher(fieldValue);
                formatIsValid = matcher.matches();
                break;
            case ALPHA:
                matcher = lettersPattern.matcher(fieldValue);
                formatIsValid = matcher.matches();
                break;
            default:
                formatIsValid = false;
        }
        return formatIsValid;
    }
}
```
